extensions/replication/tasks/CopyLocationTask.js

```diff
@@ -816,6 +816,11 @@ class CopyLocationTask extends BackbeatTask {
                     'object contents have changed');
             }
         }
+        if (objMD.getDataStoreName() === actionEntry.getAttribute('toLocation')) {
+            // The object was already transitioned to the destination location
+            return errors.InvalidObjectState.customizeDescription(
+                'object already transitioned');
+        }
```

Comment:
Maybe worth updating the function description as now it's not only the md5 but also the data store name